Tensile/KernelWriterAssembly.py

```diff
@@ -6145,17 +6151,17 @@ def mfmaIter(self, kernel, m, innerUnroll, tail=False):
         shiftK.addCode(inst("s_lshl_b32",   sgpr(tmpSgpr+2), sgpr(tmpSgpr+2), log2(shiftPerElement), "use shift to fill 0 for outside element"))
         for a in range(0, kernel["MIWaveTile"][0]):
           for iui in range(0, innerUnroll):
-            aStr  = vgpr("ValuA_X%u_I%u+%u" % (m, iui, a * vgprPerInput), vgprPerInput)
+            aStr = vgpr("ValuA_X%u_I%u+%u+%u+%u" % ((m//self.numIterPerCoalescedRead)*self.numIterPerCoalescedRead, (iui//self.numReadsIterCoalesced)*self.numReadsIterCoalesced, a*self.numReadsIterCoalesced*vgprPerInput,m%self.numIterPerCoalescedRead*kernel["InnerUnroll"]*vgprPerInput,iui%self.numReadsIterCoalesced*vgprPerInput), vgprPerInput)
             shiftK.addCode(inst("v_lshlrev_b%u" % (vgprPerInput*32), vgpr(abReg, vgprPerInput), sgpr(tmpSgpr+2), aStr, ""))
             for bk in range(0, vgprPerInput):
-              aStr  = vgpr("ValuA_X%u_I%u+%u+%u" % (m, iui, a * vgprPerInput, bk), 1)
+              aStr  = vgpr("ValuA_X%u_I%u+%u+%u+%u+%u" % ((m//self.numIterPerCoalescedRead)*self.numIterPerCoalescedRead, (iui//self.numReadsIterCoalesced)*self.numReadsIterCoalesced, a*self.numReadsIterCoalesced*vgprPerInput,m%self.numIterPerCoalescedRead*kernel["InnerUnroll"]*vgprPerInput,iui%self.numReadsIterCoalesced*vgprPerInput, bk), 1)
               shiftK.addCode(inst("v_cndmask_b32", aStr, aStr, vgpr(abReg+bk), sgpr(tmpSgpr, 2), ""))
         for b in range(0, kernel["MIWaveTile"][1]):
           for iui in range(0, innerUnroll):
-            bStr     = vgpr("ValuB_X%u_I%u+%u" % (m, iui, b*vgprPerInput), vgprPerInput)
+            bStr = vgpr("ValuB_X%u_I%u+%u+%u+%u" % ((m//self.numIterPerCoalescedRead)*self.numIterPerCoalescedRead, (iui//self.numReadsIterCoalesced)*self.numReadsIterCoalesced, b*self.numReadsIterCoalesced*vgprPerInput,m%self.numIterPerCoalescedRead*kernel["InnerUnroll"]*vgprPerInput,iui%self.numReadsIterCoalesced*vgprPerInput), vgprPerInput)
             shiftK.addCode(inst("v_lshlrev_b%u" % (vgprPerInput*32), vgpr(abReg, vgprPerInput), sgpr(tmpSgpr+2), bStr, ""))
             for bk in range(0, vgprPerInput):
-              bStr  = vgpr("ValuB_X%u_I%u+%u+%u" % (m, iui, b*vgprPerInput, bk), 1)
+              bStr = vgpr("ValuB_X%u_I%u+%u+%u+%u+%u" % ((m//self.numIterPerCoalescedRead)*self.numIterPerCoalescedRead, (iui//self.numReadsIterCoalesced)*self.numReadsIterCoalesced, b*self.numReadsIterCoalesced*vgprPerInput,m%self.numIterPerCoalescedRead*kernel["InnerUnroll"]*vgprPerInput,iui%self.numReadsIterCoalesced*vgprPerInput, bk), 1)
```

Comment:
Lots of repeated equations through here, would be clearer with some variables

Reply:
done